packages/create-ponder/src/helpers/graphqlTranslator.ts

```diff
@@ -0,0 +1,265 @@
+      const values: string[] = [];
+      definition.values?.forEach((value) => {
+        values.push(value.name.value);
+      });
+
+      enumsResult.push(
+        `	${enumName}: p.createEnum(${JSON.stringify(values)}),\n\n`,
+      );
+    }
+  });
+
+  // Record id types for ObjectTypeDefinition
+  // This is needed for references later on
+  Object.keys(schemaAST.definitions).forEach((definitionName) => {
+    const definition = schemaAST.definitions[
+      definitionName as any as number
+    ] as DefinitionNode;
+    if (definition.kind === "ObjectTypeDefinition") {
+      const tableName = definition.name.value;
+      definition.fields?.forEach((field) => {
+        const fieldName = field.name.value;
+        if (fieldName === "id") {
+          const nonNullTypeNode = field.type as NonNullTypeNode;
+          const namedTypeNode = nonNullTypeNode.type as NamedTypeNode;
+          if (!namedTypeNode?.name?.value) {
+            console.log(field);
+            throw new Error(`Unsupported id type for ${tableName}`);
+          }
+          idTypes[tableName] =
+            GraphqlTypesToPonderTypes[
+              namedTypeNode.name.value as keyof typeof GraphqlTypesToPonderTypes
+            ];
+        }
+      });
+    }
+  });
+
+  // Parse fields for ObjectTypeDefinition
+  Object.keys(schemaAST.definitions).forEach((definitionName) => {
+    const definition = schemaAST.definitions[
+      definitionName as any as number
+    ] as DefinitionNode;
+    if (definition.kind === "ObjectTypeDefinition") {
+      const tableName = definition.name.value;
+      tables[tableName] = { columns: {}, descriptions: {} };
+      definition.fields?.forEach((field) => {
+        const fieldName = field.name.value;
+        tables[tableName]!.descriptions[fieldName] = field.description?.value
+          ? `// ${field.description?.value}`
+          : "";
+        tables[tableName]!.columns[fieldName] = parseField(tableName, field);
+      });
+    }
+  });
+
+  // Assemble fields into result strings
+  Object.keys(schemaAST.definitions).forEach((definitionName) => {
+    const definition = schemaAST.definitions[
+      definitionName as any as number
+    ] as DefinitionNode;
+    if (definition.kind === "ObjectTypeDefinition") {
+      const tableName = definition.name.value;
+      const columns = tables[tableName]!.columns;
+      const descriptions = tables[tableName]!.descriptions;
+      if (
+        definition.directives?.some(
+          (directive) => directive.name.value === "entity",
+        )
+      ) {
+        result.push(
+          `
+	${tableName}: p.createTable({${Object.keys(columns)
+    .map(
+      (key) => `
+		${descriptions[key]}
+		${key}: ${columns[key]}`,
+    )
+    .join(", \n")},
+		${Object.keys(referenceFields?.[tableName] ?? {})
+      .map(
+        (key) => `
+		${key}: ${referenceFields![tableName]![key]}`,
+      )
+      .join(", \n")}
+	}),\n`,
+        );
+      } else {
+        throw new Error(
```

Comment:
Same question as above - any idea if this is reachable?

Reply:
The Graph docs had some other types other than `@entity`, but yeah it should be a skip+warning instead of throwing an error
![image](https://github.com/user-attachments/assets/a0aa5656-3d40-4d36-ac98-43dc540cc1bb)